rundeckapp/grails-app/services/rundeck/services/PasswordFieldsService.groovy

```diff
@@ -115,6 +116,43 @@ class PasswordFieldsService {
         return count
     }
 
+    public int trackPluginFields(configurations,boolean hidden, Description... descriptions) {
+        if(!configurations) {
+            return 0
+        }
+
+        int count = 0
+        int configPos = 0
+        for (config in configurations) {
+            Map props =  config.props
+            if (config == null) {
+                continue
+            }
+
+            Description desc = descriptions.find { it.name == config.type }
+            if (desc == null) {
+                configPos++
+                continue
+            }
+
+            for (property in props) {
+                if (isPluginPasswordProperty(property)) {
```

Comment:
since we have the plugin Description, we can look up the property corresponding to the config key, and determine if it is a Password value, by `isPasswordDisplay(Property)`

Reply:
There are some plugins that can use project properties whose is not implemented like a plugin property with display annotations. One example is the Service Now plugin this property:

`project.plugin.Notification.Service-Now-Notification.login=user
project.plugin.Notification.Service-Now-Notification.password=s3cret
project.plugin.Notification.Service-Now-Notification.url=https://server.service-now.com`

So, I wonder if this would be good solution to apply mask if user want even that plugin not have the password display for this property.

Reply:
I think it's better to implement it that way and create a github issue for that plugin to be upgraded to annotations.